src/app/issuer/components/badgeclass-edit-form/badgeclass-edit-form.component.ts

```diff
@@ -499,10 +479,21 @@ export class BadgeClassEditFormComponent extends BaseAuthenticatedRoutableCompon
 		}
 		this.badgeCategory = currentBadgeCategory;
 
-		// fix the issue of missing badge-frame when changing type for first time (only with safari browser)
-		// by regenerating the upload image as adjusting upload image didn't work with this issue
-		if (this.platformService.SAFARI && this.currentImage) {
-			this.generateUploadImage(this.currentImage, this.badgeClassForm.value);
+		// To update badge-frame when badge-category is changed
+		this.changeBadgeFrame();
+	}
+
+	changeBadgeFrame() {
+		if (this.imageField?.control.value) {
+			setTimeout(() => {
```

Comment:
Same here. Timeouts like this shouldn't be necessary. There *should* exist a cleaner solution to this problem

Reply:
It doesn't work without adding `timeout`. Dealing with image loading seems to be tricky sepcially with safari browser.

Reply:
I removed them after updating the code.